src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -519,6 +525,10 @@ private List<UnMappedColumnAutoMapping> createAutomaticMappings(ResultSetWrapper
     if (autoMapping == null) {
       autoMapping = new ArrayList<>();
       final List<String> unmappedColumnNames = rsw.getUnmappedColumnNames(resultMap, columnPrefix);
+      List<String> mappedInConstructorAutoMapping = constructorAutoMappingColumns.remove(mapKey);
```

Comment:
@harawata Might be useful for a comment here to understand why its removing vs others getting data.  I think I grasp it but find it somewhat confusing.

Reply:
Thank you, @hazendaz !
Added a brief comment.
The entry is used only once, so there is no point in keeping it on memory.